checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -143,5 +149,75 @@ def _build_sub_graph(self, blocks_to_render: list[int]) -> l_graph.TerraformLoca
         sub_graph.out_edges = self.local_graph.out_edges
         return sub_graph
 
+    def _create_new_resources_count(self, statement: int, main_resource: TerraformBlock) -> None:
+        for i in range(statement):
+            self._create_new_resource(main_resource, i)
+
+    def _update_count_attributes(self, attrs: dict[str, Any], new_value: int):
+        for k, v in attrs.items():
+            if isinstance(v, list) and len(v) == 1 and isinstance(v[0], str) and COUNT_KEY in v[0]:
+                attrs[k] = attrs[k][0].replace("${" + COUNT_KEY + "}", str(new_value))
+                attrs[k] = attrs[k][0].replace(COUNT_KEY, str(new_value))
+            elif isinstance(v, dict):
+                self._update_count_attributes(v, new_value)
+
+    def _update_foreach_attributes(self, attrs: dict[str, Any], new_value: str, new_key: Optional[str | int] = ''):
+        for k, v in attrs.items():
+            if isinstance(v, list) and len(v) == 1 and isinstance(v[0], str):
+                if EACH_KEY in v[0]:
+                    attrs[k][0] = attrs[k][0].replace("${" + EACH_KEY + "}", str(new_key) or str(new_value))
```

Comment:
looks like these 2 lines happen a lot, maybe extract to a method?

Reply:
Each is a bit different, and extracting to a new func that will check some other parameters to know what to do and when can be a bit confusing.